python/ray/llm/_internal/serve/configs/server_models.py

```diff
@@ -141,15 +141,15 @@ class LLMConfig(BaseModelExtended):
     )
 
     runtime_env: Optional[Dict[str, Any]] = Field(
-        None,
+        default=None,
         description=(
             "The runtime_env to use for the model deployment replica "
             "and the engine workers."
         ),
     )
 
-    model_loading_config: ModelLoadingConfig = Field(
-        description="The settings for how to download and expose the model."
+    model_loading_config: Dict[str, Any] = Field(
```

Comment:
I think we need to do a union of Dict and type here + validation. So that if ModelLoadingConfig is also passed in directly it works. validator checks the type: if the type is ModelLoadingConfig just pass through if it's dict cast and validate that the dict fields match the structure of ModelLoadingConfig.